lldb/include/lldb/API/SBDebugger.h

```diff
@@ -243,6 +245,8 @@ class LLDB_API SBDebugger {
 
   lldb::SBTarget GetDummyTarget();
 
+  void SendTelemetry(SBStructuredData *entry);
```

Comment:
Can you clarify your privacy model?

We need this  API because we want to be able to collect performance stats for our IDE debugging  layer (eg., how long each DAP packet takes, success vs failed vs timedout counts, etc).  While it's possible that the clients could have their own telemetry  outside of LLDB, it is helpful to be able to combine both client(s)' and server's telemetry data for each session 

As for privacy concerns, at least, from our POV, we ensure that the collected data is stored with restricted and auditable access. The `Destination` class, which takes care of forwarding the collected telemetry entries to its final destination, is vendor-specific. So you could choose to ignore client telemetry if it does not fit your privacy model.

Reply:
If you want to combine telemetry from LLDB and clients together, why not invert the relationship? Instead of having LLDB ingest client telemetry data and report them together, we could instead have LLDB hand its telemetry over to clients and have the client combine them.